src/game/Pet.cpp

```diff
@@ -932,6 +932,9 @@ void Pet::GivePetXP(uint32 xp)
     if (level >= maxlevel)
         return;
 
+    if (float rateXP = sWorld.getConfig(CONFIG_FLOAT_RATE_PET_XP_KILL))
```

Comment:
I think I may have actually written in that check. It was just to be on the safe side.